lib/checking_account.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'account'
+
+module Bank
+  class CheckingAccount< Account
+    attr_accessor :id, :balance
+
+    def initialize(id, balance)
+      super(id, balance)
+      @check_count = 0
+      @fee = 2
+      raise ArgumentError.new("balance must be >= -1-") if balance < 10
+    end
+
+    def withdraw(amount)
+      if amount < (@balance - 11)
```

Comment:
Would it be possible to refactor your `withdraw` methods so that the versions in the subclasses could take advantage of `Account.withdraw`? You might need to add some extra information about fees or minimum balances, but it has the potential to make the code much DRYer.

It looks like you may have been headed way with the `@fee` instance variable, but I don't think it ever gets used.